addons/sale_stock/models/stock.py

```diff
@@ -92,7 +92,7 @@ def _action_done(self):
             product = move.product_id
             so_line_vals = {
                 'move_ids': [(4, move.id, 0)],
-                'name': product.display_name,
+                'name': move.purchase_line_id and move.purchase_line_id.name or product.display_name,
```

Comment:
@sonhd91`
1. sale_stock does not depend on purchase_stock, you need to find another module, a create a new one to handle this case.
2. 'name': move.purchase_line_id.name or product.display_name,` is enough